app/components/LoginForm/RegisterForm.tsx

```diff
@@ -1,96 +1,72 @@
-    captchaResponse: string;
-  }) => any;
-} & FormProps;
-type State = {
-  submitted: boolean;
-};
+const validate = createValidator({
+  email: [required(), isEmail()],
+  captchaResponse: [required('Captcha er ikke validert')],
+});
 
-class RegisterForm extends Component<Props, State> {
-  mounted = false;
-  state = {
-    submitted: false,
-  };
+const RegisterForm = () => {
+  const [mounted, setMounted] = useState(false);
+  const [submitted, setSubmitted] = useState(false);
 
-  componentDidMount() {
-    this.mounted = true;
-  }
+  useEffect(() => {
+    setMounted(true);
+    return () => setMounted(false);
+  }, []);
 
-  componentWillUnmount() {
-    this.mounted = false;
-  }
+  const dispatch = useAppDispatch();
 
-  onSubmit = (data) => {
-    return this.props
-      .sendRegistrationEmail(data)
-      .then(() => {
-        if (this.mounted) {
-          this.setState({
-            submitted: true,
-          });
-        }
-      })
-      .catch((err) => {
-        if (this.mounted && err.payload && err.payload.response) {
-          throw new SubmissionError(err.payload.response.jsonData);
-        }
-      });
+  const onSubmit = (data) => {
+    dispatch(sendRegistrationEmail(data)).then(() => {
+      if (mounted) {
+        setSubmitted(true);
+      }
+    });
   };
 
-  render() {
-    const { handleSubmit, invalid, pristine, submitting } = this.props;
-
-    if (this.state.submitted) {
-      return (
-        <div>
-          Vi har sendt en e-post til deg hvor du kan fortsette registreringen.
-        </div>
-      );
-    }
-
+  if (submitted) {
     return (
-      <Form
-        onSubmit={handleSubmit(this.onSubmit)}
-        onClick={(e) => e.stopPropagation()}
-      >
-        <Field name="email" component={TextInput.Field} placeholder="E-post" />
-        <Field
-          name="captchaResponse"
-          fieldStyle={{
-            width: 304,
-          }}
-          component={Captcha.Field}
-        />
-        <Button dark submit disabled={invalid || pristine || submitting}>
-          Registrer deg
-        </Button>
-      </Form>
+      <div>
+        Vi har sendt en e-post til deg hvor du kan fortsette registreringen
+      </div>
     );
   }
-}
 
-const validate = createValidator({
-  email: [required(), isEmail()],
-  captchaResponse: [required('Captcha er ikke validert')],
-});
-export default compose(
-  connect<any, any, any, any, any, any>(null, {
-    sendRegistrationEmail,
-  }),
-  reduxForm({
-    form: 'RegisterForm',
-    validate,
-  })
-)(RegisterForm);
+  return (
+    <LegoFinalForm onSubmit={onSubmit} validate={validate}>
+      {({ handleSubmit }) => (
+        <Form onSubmit={handleSubmit} onClick={(e) => e.stopPropagation()}>
+          <Field
+            name="email"
+            component={TextInput.Field}
+            placeholder="E-post"
+          />
+          <Field
+            name="captchaResponse"
+            fieldStyle={{
+              width: 304,
```

Comment:
Doesn't feel too dynamic(?)

Reply:
Yup but I didn't want to bother improving css here